config.go

```diff
@@ -1,215 +1,184 @@
-	defaultLogDir         = "logs"
+	configFilename    = "./config.json"
+	defaultDebugLevel = "info"
+	defaultLogDir     = "logs"
 )
 
 var (
-	// Set default active net params.
-	activeNetParams = &params.MainNetParams
+	// spvNetParams defines the SPV module network parameters.
+	spvNetParams = &config.DefaultParams
 
-	// Load configuration from file.
-	cfg, loadConfigErr = loadNewConfig()
-)
+	// activeNetParams defines the side chain network parameters.
+	activeNetParams = &params.MainNetParams
 
-type config struct {
-	NetType       string
-	Configuration struct {
-		Magic                      uint32
-		SpvMagic                   uint32
-		SeedList                   *[]string
-		SpvSeedList                *[]string
-		ExchangeRate               float64
-		MinCrossChainTxFee         int64
-		HttpRestPort               uint16
-		HttpJsonPort               uint16
-		HttpWsPort                 uint16
-		NodePort                   uint16
-		PrintLevel                 elalog.Level
-		MaxLogsSize                int64
-		MaxPerLogSize              int64
-		FoundationAddress          string
-		DisableTxFilters           bool
-		MainChainFoundationAddress string
-		PowConfiguration           struct {
-			PayToAddr    string
-			AutoMining   bool
-			MinerInfo    string
-			MinTxFee     int64
-			InstantBlock bool
-		}
-		RpcConfiguration struct {
-			User        string
-			Pass        string
-			WhiteIPList []string
-		}
-		CheckPowHeaderHeight uint32
+	// defaultConfig defines the default configuration parameters.
+	defaultConfig = configParams{
+		DebugLevel: defaultDebugLevel,
 	}
-}
 
-type appConfig struct {
-	NodePort          uint16
-	HttpRestPort      uint16
-	HttpJsonPort      uint16
-	HttpWsPort        uint16
-	Mining            bool
-	MinerInfo         string
-	MinerAddr         string
-	LogLevel          string
-	MaxLogsFolderSize int64
-	MaxPerLogFileSize int64
-	MonitorState      bool
-	RPCUser           string
-	RPCPass           string
-	RPCWhiteList      []string
-}
+	// cfg indicates the configuration parameters load from 'config.json' file.
+	cfg = loadConfig()
+)
 
-func loadNewConfig() (*appConfig, error) {
-	appCfg := appConfig{
-		LogLevel:          defaultLogLevel,
-		MaxLogsFolderSize: defaultLogsFolderSize,
-		MaxPerLogFileSize: defaultMaxLogFileSize,
-		HttpRestPort:      20604,
-		HttpJsonPort:      20606,
-		HttpWsPort:        20605,
-		MinerAddr:         "8VYXVxKKSAxkmRrfmGpQR2Kc66XhG6m3ta",
-		MonitorState:      true,
-	}
+// configParams defines the configuration parameters of the 'config.json' file.
+type configParams struct {
+	ActiveNet          string
+	Magic              uint32
+	NodePort           uint16
+	DNSSeeds           []string
+	DisableDNS         bool
+	PermanentPeers     []string
+	SPVDNSSeeds        []string
+	SPVDisableDNS      bool
+	SPVPermanentPeers  []string
+	EnableREST         bool
+	RESTPort           uint16
+	EnableWS           bool
+	WSPort             uint16
+	DisableRPC         bool
```

Comment:
DisableRPC should be consistent with EnableREST

Reply:
The RPC service is enabled by default, and the REST and WS is disabled by default, the DisableRPC and EnableREST parameter is for different usage, it's not about consistent.

Reply:
Configuration items should be consistent.
The default startup can be achieved by setting the default value of EnableRPC to True.

Reply:
The DisableRPC is keeping consistent with the main chain configuration, of course you can change it to EnableRPC, it's depends on your choice.

Reply:
There is no EnableRPC in mainchain[release_v0.3.1].
If EnableRPC is added in the next version, it should be removed or changed to EnableRPC.